java/client/src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -4026,6 +4031,1997 @@ private Object executeProtocolCommandWithByteArgs(String commandName, byte[]...
+        try {
+            GlideString result = glideClient.hget(GlideString.of(key), GlideString.of(field)).get();
+            return result != null ? result.getBytes() : null;
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("HGET operation failed", e);
+        }
+    }
+
+    /**
+     * Sets the specified fields to their respective values in the hash stored at key. This command
+     * overwrites any specified fields already existing in the hash.
+     *
+     * @param key the key of the hash
+     * @param hash a map of field-value pairs to set in the hash
+     * @return "OK"
+     */
+    public String hmset(String key, Map<String, String> hash) {
+        checkNotClosed();
+        ensureInitialized();
+        try {
+            // Use customCommand to execute HMSET and get the actual server response
+            List<String> args = new ArrayList<>();
+            args.add("HMSET");
+            args.add(key);
+            for (Map.Entry<String, String> entry : hash.entrySet()) {
+                args.add(entry.getKey());
+                args.add(entry.getValue());
+            }
+
+            Object result = glideClient.customCommand(args.toArray(new String[0])).get();
+            return result != null ? result.toString() : null;
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("HMSET operation failed", e);
+        }
+    }
+
+    /**
+     * Sets the specified fields to their respective values in the hash stored at key (binary
+     * version).
+     *
+     * @param key the key of the hash
+     * @param hash a map of field-value pairs to set in the hash
+     * @return "OK"
+     */
+    public String hmset(final byte[] key, final Map<byte[], byte[]> hash) {
+        checkNotClosed();
+        ensureInitialized();
+        try {
+            // Use customCommand to execute HMSET and get the actual server response
+            List<String> args = new ArrayList<>();
+            args.add("HMSET");
+            args.add(new String(key));
+            for (Map.Entry<byte[], byte[]> entry : hash.entrySet()) {
+                args.add(new String(entry.getKey()));
+                args.add(new String(entry.getValue()));
+            }
+
+            Object result = glideClient.customCommand(args.toArray(new String[0])).get();
+            return result != null ? result.toString() : null;
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("HMSET operation failed", e);
+        }
+    }
+
+    /**
+     * Returns the values associated with the specified fields in the hash stored at key.
+     *
+     * @param key the key of the hash
+     * @param fields the fields in the hash
+     * @return a list of values associated with the given fields, in the same order as they are
+     *     requested
+     */
+    public List<String> hmget(String key, String... fields) {
```

Comment:
Is this signature the same in Jedis? It is odd that you can pass in an empty set of fields